app/assets/src/components/common/Heatmap/MetadataLegend.jsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+
+import { getTooltipStyle } from "~/components/utils/tooltip";
+
+import cs from "./metadata_legend.scss";
+
+export default class MetadataLegend extends React.Component {
+  render() {
+    const { columnMetadataLegend, tooltipLocation } = this.props;
+    return (
+      <div
+        className={cx(cs.tooltip, columnMetadataLegend && cs.visible)}
+        style={getTooltipStyle(tooltipLocation, {
+          buffer: 20,
+          below: true,
+        })}
+      >
+        <div className={cs.legend}>
+          {Object.keys(columnMetadataLegend)
+            .sort()
+            .map(label => (
+              <div className={cs.legendRow} key={label}>
+                <span
+                  className={cs.legendEntryColor}
+                  style={{ backgroundColor: columnMetadataLegend[label] }}
+                />
+                {label}
+              </div>
+            ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+MetadataLegend.propTypes = {
+  columnMetadataLegend: PropTypes.object,
```

Comment:
Can we rename this to metadataColors and add a comment saying "A map containing metadata values and their color in the heatmap."